eng/pipelines/templates/steps/archetype-sdk-docs.yml

```diff
@@ -23,3 +39,42 @@ steps:
           -DocGenDir ${{parameters.DocGenerationDir}}
           -ArtifactStagingDirectory '$(Build.ArtifactStagingDirectory)'
           -verbose
+
+  # artifacts will only be blank if the template is called with no Artifact parameter,
+  # which should only happen in the case of a PR build. In this case we locate all that were directly
+  # changed and build their docs.
+  - ${{ if not(parameters.Artifacts) }}:
+    - pwsh: |
+        $failedBuild = $false
+
+        $targetedPackages = Get-ChildItem -Recurse "${{ parameters.PackageInfoFolder }}" *.json
+          | ForEach-Object { Get-Content -Raw _ | ConvertFrom-Json }
+          | Where-Object { $_.IncludedForValidation -ne $true }
+
+        if ($targetedPackages) {
+          foreach($pkg in $targetedPackages) {
+            $artifactDirectory = ($pkg.DirectoryPath -split '\\|/')[-1]
+
+            &"$(Build.SourcesDirectory)/doc/ApiDocGeneration/Generate-Api-Docs.ps1" `
```

Comment:
And here re: parallel

Reply:
I will add parallelism via pshell jobs in this iteration of `net - pullrequest`. I think that the greater

> run this _entire job_ batched
that we've been discussing offline should do most of the heavy lifting beyond that though.